packages/ui/src/bounty/modals/AddBountyModal/components/WorkingDetailsStep.tsx

```diff
@@ -0,0 +1,202 @@
+  setWorkingPeriodStake: (stake: BN) => void
+  setWorkingPeriodStakeAllowance: (allowance: boolean) => void
+  whitelistLimit?: u32 & AugmentedConst<'rxjs'>
+}
+
+export const WorkingDetailsStep = ({
+  workingPeriodLength,
+  workingPeriodWhitelist,
+  workingPeriodType,
+  workingPeriodStake,
+  workingPeriodStakeAllowance,
+  setWorkingPeriodLength,
+  setWorkingPeriodStake,
+  setWorkingPeriodStakeAllowance,
+  setWorkingPeriodType,
+  setWorkingPeriodWhitelist,
+  whitelistLimit,
+}: Props) => {
+  const onMemberAdd = (member: Member) => {
+    setWorkingPeriodWhitelist([...workingPeriodWhitelist, member])
+  }
+
+  const whitelistFilter = (member: Member): boolean =>
+    !workingPeriodWhitelist.some((oldMember) => member.id === oldMember.id)
+
+  const removeMemberFromWhitelist = (member: Member) => () => {
+    setWorkingPeriodWhitelist(workingPeriodWhitelist.filter((mapMember) => mapMember.id !== member.id))
+  }
+
+  return (
+    <RowGapBlock gap={24}>
+      <Row>
+        <TextHuge bold>Working Period Details</TextHuge>
+      </Row>
+
+      <Row>
+        <InlineToggleWrap>
+          <Label>Bounty type:</Label>
+          <ToggleCheckbox
+            checked={workingPeriodType === 'open'}
+            onChange={(isSet) => (isSet ? setWorkingPeriodType('open') : setWorkingPeriodType('closed'))}
+            falseLabel={
+              <CheckBoxLabelWrapper>
+                Closed{' '}
+                <Tooltip tooltipText="Lorem ipsum...">
+                  <TooltipDefault />
+                </Tooltip>
+              </CheckBoxLabelWrapper>
+            }
+            trueLabel={
+              <CheckBoxLabelWrapper>
+                Open
+                <Tooltip tooltipText="Lorem ipsum...">
+                  <TooltipDefault />
+                </Tooltip>
+              </CheckBoxLabelWrapper>
+            }
+          />
+        </InlineToggleWrap>
+      </Row>
+
+      {workingPeriodType === 'closed' && (
+        <RowGapBlock gap={10}>
+          <TextMedium bold>Whitelist</TextMedium>
+          <TextMedium>The upper bound for whitelist is {whitelistLimit?.toHuman() || 0}.</TextMedium>
+          <InputComponent
+            disabled={+(whitelistLimit?.toHuman() || 0) === workingPeriodWhitelist.length}
+            tooltipText="Lorem ipsum dolor sit amet consectetur, adipisicing elit."
+            inputSize="l"
+          >
+            <SelectMember
+              id="select-whitelist"
+              disabled={+(whitelistLimit?.toHuman() || 0) === workingPeriodWhitelist.length}
+              filter={whitelistFilter}
+              onChange={onMemberAdd}
+            />
+          </InputComponent>
+          <WhitelistWrapper>
+            {workingPeriodWhitelist?.map((member) => (
+              <MemberWrapper key={member.id}>
+                <MemberInfo member={member} />
+                <CloseButton onClick={removeMemberFromWhitelist(member)} />
+              </MemberWrapper>
```

Comment:
I'm not sure if this is related to this PR, but when I click on Member avatar, sidebar with info about them is displayed, but AddBountyModal is closed then

Reply:
It's not connected with this PR. It could be improve in future because this changes is time consuming